app/controllers/testOnly/GoodsNotReleasedP5Controller.scala

```diff
@@ -32,20 +32,19 @@ import scala.concurrent.ExecutionContext
 class GoodsNotReleasedP5Controller @Inject() (
   override val messagesApi: MessagesApi,
   identify: IdentifierAction,
-  goodsNotReleasedActionProvider: GoodsNotReleasedActionProvider,
+  messageRetrievalAction: DepartureMessageRetrievalActionProvider,
   cc: MessagesControllerComponents,
   viewModelProvider: GoodsNotReleasedP5ViewModelProvider,
   view: GoodsNotReleasedP5View
 )(implicit val executionContext: ExecutionContext, frontendAppConfig: FrontendAppConfig)
     extends FrontendController(cc)
     with I18nSupport {
 
-  def goodsNotReleased(departureId: String, localReferenceNumber: LocalReferenceNumber, messageId: String): Action[AnyContent] =
-    (Action andThen identify andThen goodsNotReleasedActionProvider(departureId, messageId)) {
+  def goodsNotReleased(departureId: String, messageId: String): Action[AnyContent] =
+    (Action andThen identify andThen messageRetrievalAction[IE051Data](departureId, messageId)) {
```

Comment:
I wonder if this controller was added around the time the p5 check was introduced...Would you be able to add that check into the actions?